google/cloud/aiplatform/tensorboard/uploader.py

```diff
@@ -831,43 +834,79 @@ def _create_point(
         time_series_proto: tensorboard_data.TimeSeriesData,
         event: tf.compat.v1.Event,
         value: tf.compat.v1.Summary.Value,
+        metadata: tf.compat.v1.SummaryMetadata,
     ):
         """Adds a scalar point to the given tag, if there's space.
 
         Args:
           time_series_proto: TimeSeriesData proto to which to add a point.
           event: Enclosing `Event` proto with the step and wall time data.
-          value: Scalar `Summary.Value` proto with the actual scalar data.
+          value: `Summary.Value` proto.
 
         Raises:
           _OutOfSpaceError: If adding the point would exceed the remaining
             request budget.
         """
-        scalar_proto = tensorboard_data.Scalar(
-            value=tensor_util.make_ndarray(value.tensor).item()
-        )
-        point = tensorboard_data.TimeSeriesDataPoint(
-            step=event.step,
-            scalar=scalar_proto,
-            wall_time=timestamp.Timestamp(
-                seconds=int(event.wall_time),
-                nanos=int(round((event.wall_time % 1) * 10 ** 9)),
-            ),
-        )
+        point = self._create_data_point(event, value, metadata)
+
+        if not self._validate(point, event, value):
+            return
+
         time_series_proto.values.extend([point])
         try:
             self._byte_budget_manager.add_point(point)
         except _OutOfSpaceError:
             time_series_proto.values.pop()
             raise
 
+    @abc.abstractmethod
+    def _get_tracker(self):
+        """
+        :return: tracker function from upload_tracker.UploadTracker
+        """
+        pass
 
-class _TensorBatchedRequestSender(object):
-    """Helper class for building WriteTensor() requests that fit under a size limit.
+    @abc.abstractmethod
+    def _get_value_type(
```

Comment:
Based on the usage of this method, it seems like it should be an class property. ie:

```Python
class _BaseBatchRequestSender:
    ...
    @property
    @classmethod
    @abc.abstractmethod
    def _value_type(cls) -> tensorboard_time_series.TensorboardTimeSeries.ValueType:
        pass


class _ScalarBatchedRequestSender(_BaseBatchedRequestSender):
    
    _value_type = tensorboard_time_series.TensorboardTimeSeries.ValueType.SCALAR
    ....

_ScalarBatchedRequestSender._value_type
```

Reply:
awesome. Changed.